goosebit.yaml

```diff
@@ -1,32 +1,46 @@
 ## Settings to adjust for each installation
+## Settings are listed with single #, comments with 2 #
 
-# Port to host the server on, default:
+## Port to host the server on, default:
 # port: 60053 # GOOSE ;)
 
-# Database to be used, default:
+## Database to be used, default:
 # db_uri: sqlite:///<project root>/db.sqlite3
 
-# Path to the directory containing artifact files, default:
+## Path to the directory containing artifact files, default:
 # artifacts_dir: /<project root>/artifacts
 
-# Frequency that devices should check for available updates.
+## Frequency that devices should check for available updates.
 poll_time_default: 00:01:00
 
-# Frequency to log information during an update. Defines how often download progress
-# and device logs get updated.
+## Frequency to log information during an update. Defines how often download progress
+## and device logs get updated.
 poll_time_updating: 00:00:05
 
-# Whether to track the IP of the device when it polls.  Useful for debugging, but can be turned off for privacy.
+## Whether to track the IP of the device when it polls.  Useful for debugging, but can be turned off for privacy.
 # track_device_ip: true
 
-# Secret key used for parsing user sessions. It is HIGHLY advised to pass this as an environment variable instead.
-# Defaults to a randomized value. If this value is not set, user sessions will not persist when app restarts.
+## Device authentication settings
+# device_auth:
+
+##  Enable the checking of device authentication
+#   enable: false
+
+##  The auth mode, one of:
+##  - strict: Strict mode, all devices must have keys, and all keys must be set via the API.
+##  - lax: Lax mode, devices which have keys must use them, but devices without keys do not require them.
+##  - setup: Setup mode, any devices polling with an auth token that don't have one will save it.
```

Comment:
Is this the same as hawkBit implements it?

Reply:
Not actually sure, but my goal is not to be hawkBit, but to be better than hawkBit.  I think this is one of the spots where we have that flexibility to change some things, but this is also why PRs exist, I am open to comment if you guys don't agree.

Reply:
Well, in general, I opt for "as simple as possible" (KISS). IMHO pre-mature optimization should be avoided as much as possible. I would only implement a feature when there is an actual need for it.

I agree, we don't want to be hawkBit. IMHO "simplicity" should be gooseBit's main advantage. However, we should implement certain features (or a subset) in order to facilitate the migration to gooseBit.

Reply:
Agreed on the simplicity side. 

The idea was that strict mode is what most people will likely want, so that is the default when unspecified.  Lax mode maybe be a ur least useful mode, but I think the one we would use if we were to use auth mode at all.  Setup mode is basically for migration only, because it makes sure there are no issues when migrate with accidentally sending something wrong to the API.

I'm ok with the only option being strict, but I figured these options are simple and useful enough that I may as well just include them.